core/src/main/scala/org/apache/spark/sql/store/CodeGeneration.scala

```diff
@@ -260,6 +283,32 @@ object CodeGeneration extends Logging {
         "dialect")).asInstanceOf[GeneratedStatement]
   }
 
+
+  private[this] def compileGeneratedIndexUpdate(table: String,
```

Comment:
Why do we need the complexity of generated code when we are adding one row at a time? For table updates, multiple rows were being added in a single update command and iterating over them is faster with codegen. Wouldn't prepared statement have been enough in this case?  If I am not missing something basic, we should remove this whole code generation part for the index updates.